tests/unit/rovecomm_test.py

```diff
@@ -0,0 +1,306 @@
+    assert core.rovecomm.write(packet2, False) == 1
+
+    # Packet should not still be recieved because we unsubscribed
+    time.sleep(.05)
+    assert responses.get(4233, (1, 2, 3)) == (1, 2, 3)
+
+
+def test_old_version_tcp():
+    global responses
+    # 5 is the data id for the invalid version return packet
+    core.rovecomm.set_callback(5, handle_packet)
+
+    # Test socket to try to send to RoveComm over TCP
+    s = socket.socket(socket.AF_INET, socket.SOCK_STREAM)
+    s.connect(("127.0.0.1", 11111))
+    rovecomm_packet = struct.pack(">BHBB", 1, 4232, 2, 0)
+    data = (1, 7)
+    for i in data:
+        rovecomm_packet = rovecomm_packet + struct.pack('>b', i)
+
+    s.send(rovecomm_packet)
+
+    # Give the listener thread a moment to catch the packet
+    time.sleep(.05)
+    assert responses[5].data == (1,)
+    assert responses[5].data_type == 'b'
+    assert responses[5].data_count == 1
+    assert responses[5].data_id == 5
+
+
+def test_old_version_udp():
+    global responses
+    responses.pop(5, None)
+    # 5 is the data id for the invalid version return packet
+    core.rovecomm.set_callback(5, handle_packet)
+
+    # Test socket to try to send to RoveComm over UDP
+    s = socket.socket(socket.AF_INET, socket.SOCK_DGRAM)
+    s.bind(("", 11001))
+
+    rovecomm_packet = struct.pack(">BHBB", 1, 4231, 2, 0)
+    data = (1, 6)
+    for i in data:
+        rovecomm_packet = rovecomm_packet + struct.pack('>b', i)
+
+    s.sendto(rovecomm_packet, ('127.0.0.1', 11000))
+
+    # Give the listener thread a moment to catch the packet
+    time.sleep(.05)
+    assert responses[5].data == (1,)
+    assert responses[5].data_type == 'b'
+    assert responses[5].data_count == 1
+    assert responses[5].data_id == 5
+
+
+def test_read_exception_udp():
+    global responses
+    # 0 is the data id for the default packet, sent when exception is raised
+    core.rovecomm.set_callback(0, handle_packet)
+
+    # A mock socket to throw an exception when a read occurs
+    mock = Mock()
+    mock.recvfrom.side_effect = Exception
+    temp = core.rovecomm.udp_node.RoveCommSocket
+    mock.fileno.return_value = temp.fileno()
+    core.rovecomm.udp_node.RoveCommSocket = mock
+
+    # Send a packet so recv can be triggered
+    s = socket.socket(socket.AF_INET, socket.SOCK_DGRAM)
+    s.bind(("", 11001))
+
+    rovecomm_packet = struct.pack(">BHBB", 1, 4230, 2, 0)
+    data = (1, 6)
+    for i in data:
+        rovecomm_packet = rovecomm_packet + struct.pack('>b', i)
+
+    s.sendto(rovecomm_packet, ('127.0.0.1', 11000))
+
+    # Give the listener thread a moment to catch the packet
+    time.sleep(.05)
+    # Match the packet to the blank packet
+    assert responses[0].data == ()
+    assert responses[0].data_type == 'b'
+    assert responses[0].data_count == 0
+    assert responses[0].data_id == 0
+
+    # Gives the udp node its socket back
+    core.rovecomm.udp_node.RoveCommSocket = temp
+
+
+def test_listener_shutdown():
```

Comment:
hmmm, this will cause issues in other unit tests if we rely on core.rovecomm having a listening thread. It should be done, we will just need to adjust future tests accordingly.